src/plugins/SoundPlugin.js

```diff
@@ -0,0 +1,414 @@
+    this.sfxButton = document.querySelector(sfxButton);
+    this.voButton = document.querySelector(voButton);
+
+    if (this.soundButton) {
+      this.soundButton.addEventListener('click', this.onSoundToggle.bind(this));
+    }
+
+    if (this.musicButton) {
+      this.musicButton.addEventListener('click', this.onMusicToggle.bind(this));
+    }
+
+    if (this.sfxButton) {
+      this.sfxButton.addEventListener('click', this.onSFXToggle.bind(this));
+    }
+
+    if (this.voButton) {
+      this.voButton.addEventListener('click', this.onVOToggle.bind(this));
+    }
+
+    if (this.soundSlider) {
+      this.soundSlider.addEventListener(
+        'change',
+        this.onSoundVolumeChange.bind(this)
+      );
+    }
+
+    if (this.musicSlider) {
+      this.musicSlider.addEventListener(
+        'change',
+        this.onMusicVolumeChange.bind(this)
+      );
+    }
+
+    if (this.sfxSlider) {
+      this.sfxSlider.addEventListener(
+        'change',
+        this.onSfxVolumeChange.bind(this)
+      );
+    }
+
+    if (this.voSlider) {
+      this.voSlider.addEventListener(
+        'change',
+        this.onVoVolumeChange.bind(this)
+      );
+    }
+
+    this.client.on(
+      'features',
+      function(features) {
+        if (
+          !features.data ||
+          'object' !== typeof features.data ||
+          null === features.data
+        ) {
+          return;
+        }
+        if (this.voButton instanceof HTMLElement) {
+          this.voButton.style.display = features.data.vo
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.musicButton instanceof HTMLElement) {
+          this.musicButton.style.display = features.data.music
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.soundButton instanceof HTMLElement) {
+          this.soundButton.style.display = features.data.sound
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.sfxButton instanceof HTMLElement) {
+          this.sfxButton.style.display = features.data.sfx
+            ? 'inline-block'
+            : 'none';
+        }
+
+        if (this.soundSlider) {
+          this.soundSlider.style.display = features.data.soundVolume
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.voSlider) {
+          this.voSlider.style.display = features.data.voVolume
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.musicSlider) {
+          this.musicSlider.style.display = features.data.musicVolume
+            ? 'inline-block'
+            : 'none';
+        }
+        if (this.sfxSlider) {
+          this.sfxSlider.style.display = features.data.sfxVolume
+            ? 'inline-block'
+            : 'none';
+        }
+      }.bind(this)
+    );
+  }
+
+  /**
+   *
+   *
+   * @memberof SoundPlugin
+   */
+  onSoundVolumeChange() {
+    this.soundVolume = this.volumeRange(Number(this.soundSlider.value));
```

Comment:
Potential question here: Does `volume = 0` the same as muted? We'll potentially need to keep the muted flag and volume levels in sync. Moreover, we'll also need to keep button states and slider positions synced as well.

Reply:
volume and muted are technically different, but when setting the volume to 0 will set the muted state to true for that channel